src/sentry/snuba/discover.py

```diff
@@ -634,8 +644,24 @@ def top_events_timeseries(
             limit=10000,
             referrer=referrer,
         )
+        if len(top_events["data"]) == limit and len(result.get("data", [])) and include_other:
+            other_result = raw_query(
+                aggregations=snuba_filter.aggregations,
+                conditions=other_conditions,
+                filter_keys=snuba_filter.filter_keys,
+                start=snuba_filter.start,
+                end=snuba_filter.end,
+                rollup=rollup,
+                orderby=["time"],
+                groupby=["time"],
+                dataset=Dataset.Discover,
+                limit=10000,
+                referrer=referrer + ".other",
+            )
+        else:
+            other_result = {"data": []}
 
-    if not allow_empty and not len(result.get("data", [])):
+    if not allow_empty and not len(result.get("data", [])) + len(other_result.get("data", [])):
```

Comment:
Nit: the order of operations here is kind of hard to read. Perhaps something like

```suggestion
    if (
        not allow_empty
        and not len(result.get("data", []))
        and not len(other_result.get("data", []))
    ):
```